.devcontainer/devcontainer.json

```diff
@@ -23,7 +23,7 @@
 		"njpwerner.autodocstring",
 		"zxh404.vscode-proto3",
 	],
-	"image": "ghcr.io/magma/magma/devcontainer:sha-e05d447",
+	"image": "ghcr.io/magma/magma/devcontainer:master",
```

Comment:
How frequently would this value update?

Reply:
_TLDR:_ Not often (or not at all).

_Long:_ We noticed that with our late interest in devContainer improvements, we need to update the image (sha) quite often to make the improvements available. Also there have been cases, where the Bazel team had additional requirements or was impeded, due to some changes not getting in the "production devcontainer". Thus we wanted to go towards a more rolling release scheme for the devcontainer image for at least as long as we are frequently improving it.
If desired, we can easily go back to a sha based manual approach by reverting #11508, once frequent changes are no longer required.

We might want to discuss some more CI improvements in order to stay with a rolling release scheme for devcontainer, which I would see beneficial though.

Reply:
@Neudrino If this means that the devcontainer gets rebuilt only when the underlying is updated, then I'm ok with that. (That shouldn't be happening too often anyways).

Reply:
@themarwhal, sorry, but I am not sure I completely get what you are asking. :sweat_smile: 

A rebuild of the container is defined by [the Github Action](https://github.com/magma/magma/blob/master/.github/workflows/docker-builder-devcontainer.yml), of course. However, this did not change to before. I.e. it rebuilds on schedule every 6 hours as well as on any change made on master to the respective files. (If the rebuild every 6 hours is sensible is probably debatable but out of scope here.)

If you are asking, when functional changes will occur, than:
- if a reviewed merge to master happens, or
- if the base image changes (assuming, we keep the scheduled rebuild).

Reply:
Ah sorry! I should have been more clear. I'm not sure if you've seen this, but if you have a codespace open and rebase ontop of a commit (which is often master) that has a different underlying codespace image or configuration, you get a little pop up that says whether the user would like to rebuild the devcontainer. This is great if the container has actually been updated, but not too great if not. (The rebuild takes about a few minutes). 

So I just wanted to avoid GitHub codespaces rebuilding the dev env everytime a new commit is pushed. 

I do not know this will happen with this change though, just raising my concerns. I also think we probably do not have to rebuild the image every 6 hours. :) So maybe this is a way to counter this too.

Reply:
cc @Neudrino

Reply:
Great. This sounds good to me. cc @electronjoe for visibility

Reply:
Anyway, if this turns out to deviate from the expectations, we can easily make a small change to revert the behaviour. :wink:

Reply:
Lets give it a try!